module/zfs/dbuf.c

```diff
@@ -3269,178 +3241,111 @@ dbuf_hold_copy(struct dbuf_hold_arg *dh)
 	}
 
 	rw_enter(&db->db_rwlock, RW_WRITER);
-	bcopy(data->b_data, db->db.db_data, arc_buf_size(data));
+	bcopy(dr->dt.dl.dr_data->b_data, db->db.db_data, db->db.db_size);
```

Comment:
I believe you do want to use `arc_buf_size(data)` here.

Reply:
Right, I will use `arc_buf_size(data)`.